editor/src/clj/editor/app_view.clj

```diff
@@ -2697,36 +2698,37 @@ If you do not specifically require different script states, consider changing th
-                                            commands
-                                            options
-                                            project
-                                            changes-view
-                                            (fn [successful]
-                                              (if successful
-                                                (future/complete! f nil)
-                                                (future/fail! f (LuaError. "Bob invocation failed")))
-                                              (.close out)))
+                   (let [f (future/make)]
+                     (fx/on-fx-thread
+                       (let [options (cond-> options
+                                             (not (contains? options "build-server"))
+                                             (assoc "build-server" (native-extensions/get-build-server-url prefs project evaluation-context))
+                                             (not (contains? options "build-server-header"))
+                                             (assoc "build-server-header" (native-extensions/get-build-server-headers prefs)))
+                             main-scene (g/node-value app-view :scene evaluation-context)
+                             tool-tab-pane (g/node-value app-view :tool-tab-pane evaluation-context)
+                             render-build-error! (make-render-build-error main-scene tool-tab-pane build-errors-view)
+                             render-reload-progress! (make-render-task-progress :resource-sync)
+                             render-save-progress! (make-render-task-progress :save-all)
+                             render-build-progress! (make-render-task-progress :build)
+                             task-cancelled? (make-task-cancelled-query :build)
+                             out (start-new-log-pipe!)]
+                         (build-errors-view/clear-build-errors build-errors-view)
+                         (disk/async-bob-build! render-reload-progress!
+                                                render-save-progress!
+                                                render-build-progress!
+                                                out
+                                                task-cancelled?
+                                                render-build-error!
+                                                commands
+                                                options
+                                                project
+                                                changes-view
+                                                (fn [successful]
+                                                  (if successful
+                                                    (future/complete! f nil)
+                                                    (future/fail! f (LuaError. "Bob invocation failed")))
+                                                  (.close out)))))
```

Comment:
Mostly indentation. I wrapped the whole command in `fx/on-fx-thread`. The problem was that `build-errors-view/clear-build-errors` could be invoked from the background thread, and that could cause the error `java.lang.IllegalStateException: Not on FX application thread; currentThread = ForkJoinPool.commonPool-worker-37`